pkg/nodeconfig/payload/payload.go

```diff
@@ -110,6 +110,36 @@ param(
     }
   }
 
+# This retries getting the HNS-network until it succeeds
+function Retry-GetHnsNetwork {
+     param (
+         [int]$retryCount = 20,
+         [int]$retryDelaySeconds = 10
+     )
+ 
+     $attempt = 1
+ 
+     while ($attempt -le $retryCount) {
+         try {
+             $hns_network = Get-HnsNetwork | Where-Object { $_.Name -eq 'HNS_NETWORK' }
+             
+             # Check if hns_network is null
+             if ($null -eq $hns_network) {
+                 Write-Host "Attempt $($attempt + 1) returned null. Retrying in $retryDelaySeconds seconds..."
```

Comment:
Not following here, why keeping the arithmetic operation on each iteration?

Reply:
See https://github.com/openshift/windows-machine-config-operator/pull/2733/commits/da87782443049b253879f43331ef53787eee139d#r1983512639